structs_marshal.go

```diff
@@ -216,12 +216,25 @@ func (l *LogFilter) MarshalJSON() ([]byte, error) {
 	}
 
 	v := a.NewArray()
-	for indx, topic := range l.Topics {
-		if topic == nil {
+	for indx, topics := range l.Topics {
+		if topics == nil {
 			v.SetArrayItem(indx, a.NewNull())
-		} else {
-			v.SetArrayItem(indx, a.NewString(topic.String()))
+
+			continue
 		}
+
+		innerTopicArray := a.NewArray()
```

Comment:
There is none right now but could you create a file `structs_marshal_test.go` and test that the output is correct? Otherwise, I will jump on the branch and do it myself if you do not have bandwidth.

Reply:
Added coverage tests 👍 


1a5279d